vllm/attention/ops/prefix_prefill.py

```diff
@@ -675,24 +696,49 @@ def context_attention_fwd(q,
                               k,
                               v,
                               o,
+                              kv_cache_dtype: str,
                               k_cache,
                               v_cache,
                               b_loc,
                               b_start_loc,
                               b_seq_len,
                               b_ctx_len,
                               max_input_len,
+                              fp8_k_scale: float = 1.0,
+                              fp8_v_scale: float = 1.0,
                               alibi_slopes=None,
                               sliding_window=None):
 
         cap = current_platform.get_device_capability()
         BLOCK = 128 if cap[0] >= 8 else 64
+        NUM_WARPS = 8
 
         # need to reduce num. blocks when using fp32
         # due to increased use of GPU shared memory
         if q.dtype is torch.float32:
             BLOCK = BLOCK // 2
 
+        # Conversion of FP8 Tensor from uint8 storage to
+        # appropriate torch.dtype for interpretation by Triton
+        if "fp8" in kv_cache_dtype:
+            assert (k_cache.dtype == torch.uint8)
+            assert (v_cache.dtype == torch.uint8)
+
+            if kv_cache_dtype == "fp8" or kv_cache_dtype == "fp8_e4m3":
+                target_dtype = torch.float8_e4m3fn
+            elif kv_cache_dtype == "fp8_e5m2":
+                target_dtype = torch.float8_e5m2
+            else:
+                raise ValueError("Unsupported FP8 dtype:", kv_cache_dtype)
+
+            k_cache = k_cache.view(target_dtype)
+            v_cache = v_cache.view(target_dtype)
+
+        if (k_cache.dtype == torch.uint8
+                or v_cache.dtype == torch.uint8 and kv_cache_dtype == "auto"):
+            raise ValueError("kv_cache_dtype='auto' unsupported for\
+                FP8 KV Cache prefill kernel")
+
```

Comment:
Maybe we could let k_cache and v_cache in real type, and remove the conversion and checking.

Here are the discussions about kv cache dtype in https://github.com/vllm-project/vllm/pull/3234#discussion_r1522561127.

Reply:
No. Other kernels depend on uint8

Reply:
Which kernel?

Reply:
Other code may depend implicitly on this behaviour. This refactor to store fp8 KV cache not in uint8 dtype is unrelated to this PR and prefer if it is done separately.

Reply:
Created a PR here: https://github.com/vllm-project/vllm/pull/7302